### ISSUE TITLE
Option to hide name and address fields on checkout

### ISSUE THREAD

**Issue by pderksen** [easydigitaldownloads/Easy-Digital-Downloads#1040]
I've been doing some more research on checkout form conversion and I don't think name or address is required for payment. At least not for PayPal and Stripe (the 2 I use).

If I'd rather not store the user's name or register them (I currently don't register them), that's a lot less fields for them to fill out, which usually reduces the work it takes to fill out the checkout form.

For PayPal all it needs is an email address to continue right? Why not just ask for that?

Stripe shows examples of combining the first and last name into a field at the very least.

![stripe_checkout_popup_2013-03-27_2223](https://f.cloud.github.com/assets/79667/311989/ba3022f6-9767-11e2-8f4d-6d846318291f.png)

But I believe name isn't even required for Stripe or many CC merchants. See Hyperink for example.

![hyperink_minimal_checkout_2013-03-27_2215](https://f.cloud.github.com/assets/79667/311991/d27c161c-9767-11e2-8c3a-a8474bfa9fbf.png)

TL;DR:
- Add option to hide and/or make address fields required
- Add option to hide and/or make name fields required
- Add option to combine first and last name into a single field
